### ISSUE TITLE
Clarification regarding statefulness of the backup system

### ISSUE THREAD

**Issue by errge11ce** [github/backup-utils#146]
I assume that all the state of the backup system is stored in the data directory of the backup-utils, but I would like this to be clarified, since we plan to have TWO backups for the same system.

So two fully independent machines will be periodically calling ghe-backup from cron and I want to make sure that this is an OK design and the GHE instance or any of the backup instances will not be confused with this setup.

It would be nice to have this information after reading through the README.

**Comment by dbussink** [comment]
This is actually not safe. The problem is that in order to backup Git repositories with `rsync` safely, it's needed to disable `git gc` operations while the backup is in progress. You can see this process documented in https://github.com/github/backup-utils/blob/master/share/github-backup-utils/ghe-backup-repositories-rsync#L9-L32. 

When running backups from two locations, the problem is that this introduces a race condition for this part. If the two backups would run at the same time, it means that one of the backups could also be finished and remove the `.sync_in_progress` file. This means the second backup that then continues to run might have a concurrent `git gc` operation, which can lead to inconsistent snapshots. 

The changes of this happening are not super high and it will look like it works most of the time, but we can't guarantee the consistency of the backup when two backups happen at the same in parallel. If you need to have backups in two places, I'd recommend making a backup on one system and then `rsync` that to another location after the backup finishes successfully. 

I'll see if it makes sense to add an explicit mention about this in the `README`.

**Comment by errge11ce** [comment]
This recommendation is of course totally against normal security policies, as the point of the two backups is exactly that they are independent.  If someone compromises the first backup machine and then replaces all the files with fake data and then waits a day to make sure that my secondary rsync caught up, then after that I have no backup at all.

Anyways, it would be nice if this could be fixed in the future, but thanks for the information, we will plan accordingly for now.

Cheers!
